### PR TITLE
Created list of examples for available features and metrics

### PR DESCRIPTION
**Pull request by MarcCote**
This PR adds a new example file that will eventually contains more advanced usages related to QuickBundles and the clustering framework. For now, there is only one example about how to use `ResampleFeature` to manually specify the number of points that streamlines will have when used by the QuickBundles algorithm.

This PR also fixes some typos that were present in the `segment_quickbundles.py` example.

### CODE REVIEW

**Comment by arokem** [comment]
Is this work in progress? Are you planning to add more text/code in this example?

**Comment by MarcCote** [comment]
No, it is ready to be merged. I just wanted to fix the current tutorial about QuickBundles ans show how to change the number of points in the resampling.

File: doc/examples/segment_quickbundles_advanced.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+from nibabel import trackvis as tv
+from dipy.segment.clustering import QuickBundles
+from dipy.data import get_data
+
+fname = get_data('fornix')
+streams, hdr = tv.read(fname)
+streamlines = [i[0] for i in streams]
+
+"""
+QuickBundles using `ResampleFeature`
```
Comment (Garyfallidis) [inline doc/examples/segment_quickbundles_advanced.py]: the

**Comment by arokem** [comment]
OK - my impression is that this would work well as an addition to the clustering framework tutorial, rather than its own thing. As it is, it feels like it requires some context, and that other (excellent) tutorial provides that context. In other words, I doubt that anyone would ever read this on its own anyway, so why not integrate it with the other material more fluidly?

**Comment by MarcCote** [comment]
I had in mind that every extensions of `Feature` and `Metric` in dipy should have a small tutorial/example. I was afraid that it might be too much to put everything in what I consider a basic example of how to use QuickBundles.
Maybe **advanced** was not the right term maybe **additional examples**?

**Comment by arokem** [comment]
A matter of taste, I guess. For now, this tutorial doesn't contain multiple
additional examples, only this one about resampling. That's fine, but then
the title should really be something about resampling. And again, it would
be good if it contained some explanation about when/why you might want to
use resampling.

**Comment by MarcCote** [comment]
Ok, I see your point. I'll try to add more examples this week. I'll also add more context about why one would be interested to use different Features or Metrics.

**Comment by arokem** [comment]
Cool. Feel free to ping me when you're done with that, so I can take a look.

**Comment by arokem** [comment]
Hey @MarcCote - just checking in about this. Have you had a chance to take another look?

**Comment by MarcCote** [comment]
@arokem you can have a look. That is more what I had in mind. Right now, only the `Feature`s examples are done but I'll do the same for `Metric`.

**Comment by arokem** [comment]
Nice! How about combining both of these into one tutorial? Too much?

**Comment by MarcCote** [comment]
@arokem you mean combining features and metrics tutorials? I'm afraid that as we will integrate more features and metrics, it will get too big. What do you think @Garyfallidis ?

**Comment by arokem** [comment]
One option is to list the different features and metrics that can be used (and how they are combined), but not explicitly write out lines of code for each one.

**Comment by arokem** [comment]
Any thoughts here @Garyfallidis ?

**Comment by MarcCote** [comment]
I think it is nice to have a working, independent, example for each feature and metric. Personally, when I follow a tutorial I like to be able to copy-paste the code in a IPython notebook and play with it (that is why each example have its imports).

I don't really mind putting metrics and features examples in one "tutorial", even if I don't think it is necessary. I mean it is not like we are trying to minimize the number of tutorial pages.

I don't have time right now to write the example for the metrics. If you really want to merge this PR, I can remove the metrics part and make another PR later on.

**Comment by Garyfallidis** [comment]
Hello both. I think what @MarcCote suggests having 3 tutorials. A very basic for QB and then one for metrics and one for features makes perfect sense to me. So let's move towards that direction. But of course interconnect the tutorials.

Also on copying the imports on different sections of this tutorials. I think this is an exception of the classic tutorials that we write because here Marc is listing a series of features and he wants to make them copyable at any stage and order. So, I am fine with repeating the imports for this case. @arokem if you see the rendered tutorial it will be really clear. So, no strong feelings here. But I think it's fine to do what Marc proposes.

**Comment by arokem** [comment]
Sure - no problem

File: doc/examples/segment_clustering_features.py [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+**When:** The QuickBundles algorithm requires streamlines to have the same
+number of points. By default, QuickBundles uses `ResampleFeature` to resample
+streamlines so that they have 12 points each [Garyfallidis12]_. If you want to
+use a different number of points for the resampling, you should provide your
+own instance of `ResampleFeature` (see following example).
+
+**Note:** Resampling streamlines has an impact on clustering results both in
+term of speed and quality. Setting the number of points too low will result in
+a loss of information about the shape of the streamlines. On the contrary,
+setting the number of points to high will slow down the clustering process.
```
Comment (Garyfallidis) [inline doc/examples/segment_clustering_features.py]: too

File: doc/examples/segment_clustering_features.py [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+
+qb = QuickBundles(threshold=5., metric=metric)
+clusters = qb.cluster(streamlines)
+
+# Extract feature of every streamline.
+midpoints = np.asarray(map(feature.extract, streamlines))
+
+# Color each midpoint according to the cluster they belong to.
+rng = np.random.RandomState(42)
+#colormap = rng.rand(len(clusters), 3)
```
Comment (Garyfallidis) [inline doc/examples/segment_clustering_features.py]: space after #

Comment (Garyfallidis) [inline doc/examples/segment_clustering_features.py]: or remove comment if not needed

File: doc/examples/segment_clustering_features.py [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+
+   **Showing the middle point of each streamline and colored according to the
+   QuickBundles results**.
+
+.. _clustering-examples-ArcLengthFeature:
+
+ArcLength Feature
+=================
+**What:** Instances of `ArcLengthFeature` compute the length of a streamline.
+More specifically, this feature corresponds to the sum of every streamline
```
Comment (Garyfallidis) [inline doc/examples/segment_clustering_features.py]: to the sum of the lengths of ...

File: doc/examples/segment_clustering_features.py [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+
+# Get some streamlines.
+streamlines = get_streamlines()  # Previously defined.
+
+feature = ArcLengthFeature()
+metric = EuclideanMetric(feature)
+qb = QuickBundles(threshold=2., metric=metric)
+clusters = qb.cluster(streamlines)
+
+#lengths = map(feature.extract, streamlines)
```
Comment (Garyfallidis) [inline doc/examples/segment_clustering_features.py]: pep8 or remove

**Comment by MarcCote** [comment]
I guess this will fix #715 and close #719

File: doc/examples/segment_clustering_metrics.py [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+return the average of those distances. This metric takes as inputs two features
+that are sequences containing the same number of elements.
+
+**When:** By default the `QuickBundles` clustering will resample your
+streamlines on-the-fly so they have 12 points. If for some reason you want
+to avoid this and you made sure all your streamlines have already the same
+number of points, you can manually provide an instance of
+`AveragePointwiseEuclideanMetric` to `QuickBundles`. Since the default
+`Feature` is the `IdentityFeature` the streamlines won't be resampled thus
+saving some computional time.
```
Comment (Garyfallidis) [inline doc/examples/segment_clustering_metrics.py]: computational

File: doc/examples/segment_clustering_metrics.py [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+import numpy as np
+from dipy.viz import fvtk
+from dipy.segment.clustering import QuickBundles
+from dipy.segment.metric import VectorBetweenEndpointsFeature
+from dipy.segment.metric import CosineMetric
+
+# Get some streamlines.
+streamlines = get_streamlines()  # Previously defined.
+
+feature = VectorBetweenEndpointsFeature()
```
Comment (Garyfallidis) [inline doc/examples/segment_clustering_metrics.py]: I am suggesting an alternative name VectorOfEndpointsFeature

Comment (Garyfallidis) [inline doc/examples/segment_clustering_metrics.py]: It's shorter ...

**Comment by Garyfallidis** [comment]
Nice! Good job!

**Pull request merged**